react/features/base/util/httpUtils.js

```diff
@@ -1,15 +1,25 @@
+import { timeoutPromise } from './timeoutPromise';
+
 const logger = require('jitsi-meet-logger').getLogger(__filename);
 
+/**
+ * The number of milliseconds before deciding that we need retry a fetch request.
+ *
+ * @type {number}
+ */
+const RETRY_TIMEOUT = 3000;
+
 /**
  * Wrapper around fetch GET requests to handle json-ifying the response
  * and logging errors.
  *
  * @param {string} url - The URL to perform a GET against.
+ * @param {?boolean} retry - Whether the request will be retried after short timeout.
  * @returns {Promise<Object>} The response body, in JSON format, will be
  * through the Promise.
  */
-export function doGetJSON(url) {
-    return fetch(url)
+export function doGetJSON(url, retry) {
+    const fetchPromise = fetch(url)
```

Comment:
What this method does below is confusing t ome. Why is it in the jsonify that it's doing Promise.reject on successful jsonify?

Reply:
Cause the response was not successful, did not return 200 OK.

Reply:
Okay, I think that's confusing and could easily be cleaned up by moving jsonify its own chained call, that way there is always one return, but you didn't change it so we'll let it be.

Reply:
Damn, I wrote this mess. What the hell is wrong with my dumbass?